app/views/coronavirus_landing_page/components/landing_page/_postcode-lookup.html.erb

```diff
@@ -0,0 +1,42 @@
+<%
+  input_label ||= "Enter your postcode"
+  input_value ||= ""
+  input_error ||= ""
+  hint_text ||= "For example SW1A 2AA"
+  button_text ||= "Find"
+  error_message ||= nil
+  error_description ||= nil
+  autocomplete ||= "postcode"
+  link ||= { text: "Find a postcode on Royal Mail's postcode finder", href: "http://www.royalmail.com/find-a-postcode"}
+%>
+<% if error_message %>
+  <%= render "govuk_publishing_components/components/error_alert", {
+      message: error_message,
+      description: error_description,
+    } %>
+<% end %>
+<div class="covid__postcode-lookup">
+  <form action="" method="post">
```

Comment:
@leenagupte @Rosa-Fox - do we have a route to post this to yet or are you happy for us to update this once you do?

Reply:
@DilwoarH Should the form action be here? Shouldn't it be in the view that calls this partial?

Reply:
Also, I'm not sure about the location of this view. I don't think "landing page" is an appropriate place to put it as the lookup may not end up on the landing page. Could it go in "shared" instead?

Reply:
All of the code for the postcode will be within this partial, this will include the form. We've included the components needed for the rest of the pattern within this partial.

Reply:
I think shared makes more sense than landing page as this is something that will potentially be used for hub pages too

Reply:
I'd pictured this being a bit more reusable, so being able to be rendered on any page in collections, that's why I'm surprised that the form has been included as well.

Reply:
@leenagupte Reusable in what context? Only within `collections` but not within other frontend repos? Is reusable in scope of for MVP?

Reply:
Reusable within this app.

Reply:
It's not a component, but some of the same principles could/should still apply: accessible, testable, isolated..
So, perhaps rather than "resuable" I thought it would be isolated from the coronavirus specific code.

Reply:
So `collections` app only?  Would that mean it wouldn't be able be reused elsewhere? Reusable in the context of only a _postcode_-look up or all types of possible lookup? 

As this pattern appears on multiple places (eg `find-my-council`) it might make sense to eventually add this as as a component (or pattern) so it can be reused easily.  However this is probably a wider conversation outside the scope of MVP / `collections` only.

Reply:
Yes, the collections app only, I'm not talking about wider use or considering other postcode lookups that exist elsewhere. But the partial can still follow the principle of being "isolated" within this app.

Reply:
> I'd pictured this being a bit more reusable, so being able to be rendered on any page in collections, that's why I'm surprised that the form has been included as well.

I think as it is we can reuse it - I think the only change that will be needed if we do use it in other places is moving it to a more generic place within the app.

We could move the form outside of the partial but I don't think without it, it has any purpose. ie. it will just be an input and button.

Happy to discuss if you are thinking about something I haven't answered.

Reply:
> I think as it is we can reuse it - I think the only change that will be needed if we do use it in other places is moving it to a more generic place within the app.

> We could move the form outside of the partial but I don't think without it, it has any purpose. ie. it will just be an input and button.

Agreed, if this was used elsewhere it would be moved out of the folder.  Yeah it would be good to discuss, I'd like to understand what possible other use-cases (_aside_ from a postcode lookup) might this be used for within the context of this particular app. (eg last name lookup?) Also how does this link into the scope of MVP delivery.

> I'm not talking about wider use or considering other postcode lookups that exist elsewhere. But the partial can still follow the principle of being "isolated" within this app.

It might be worth considering to be added to the [Component Guide](https://components.publishing.service.gov.uk/component-guide) as it appears this might be have the isolated, reusable properties described rather that investment into just the `collections app` only, especially as other implementations exists on other repos.

Reply:
So when I was taking about reusability, I was thinking about it as a "postcode lookup" but for other things (if need arose). i.e. that submit to a different form, but all still with collections.